File: src/DeployManager.Domain/Entities/DeployRule.cs
using DeployManager.Domain.Common;

namespace DeployManager.Domain.Entities;

public class DeployRule : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string SourcePattern { get; set; } = string.Empty;
    public string DestinationPath { get; set; } = string.Empty;
    public string Action { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool IsActive { get; set; } = true;
    public Guid? DeployRuleSetId { get; set; }
    public DeployRuleSet? DeployRuleSet { get; set; }
}

File: src/DeployManager.Client/src/shared/types/deployRule.ts
export interface DeployRule {
  id: string
  name: string
  sourcePattern: string
  destinationPath: string
  action: string
  order: number
  isActive: boolean
  createdAt: string
  updatedAt: string | null
}

export interface CreateDeployRuleRequest {
  name: string
  sourcePattern: string
  destinationPath: string
  action: string
  order: number
  isActive?: boolean
}

export interface UpdateDeployRuleRequest extends CreateDeployRuleRequest {
  id: string
  isActive: boolean
}
